### PR TITLE
add tracks api with proper relationships

### LINKED ISSUE #3707: Implement tracks api with proper permissions and relationships

### PR DESCRIPTION
**Pull request by shubham-padia**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
add tracks api with proper relationships

#### Changes proposed in this pull request:

- /tracks/<int:id>/event
- /events/<int:event_id>/tracks
- /events/<int:id>/relationships/tracks
- /tracks/<int:track_id>/sessions
- /sessions/<int:id>/relationships/track
- /events/<int:event_id>/tracks
- /sessions/<int:id>/track
- /tracks
- /tracks/<int:id>
- /tracks/<int:id>/relationships/sessions
- /tracks/<int:id>/relationships/event

Fixes #3707

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -26,11 +27,12 @@
 api.route(EventList, 'event_list', '/events')
 api.route(EventDetail, 'event_detail', '/events/<int:id>', '/tickets/<int:ticket_id>/event',
           '/microlocations/<int:microlocation_id>/event', '/social_links/<int:social_link_id>/event',
-          '/sponsors/<int:sponsor_id>/event')
+          '/sponsors/<int:sponsor_id>/event', '/tracks/<int:id>/event')
```
Comment (SaptakS) [inline app/api/__init__.py]: Shouldn't this be track_id?

File: app/api/events.py [hunk 2]
```diff
@@ -45,6 +45,14 @@ class Meta:
                                schema='SocialLinkSchema',
                                many=True,
                                type_='social_link')
+    tracks = Relationship(attribute='track',
+                         self_view='v1.event_tracks',
```
Comment (SaptakS) [inline app/api/events.py]: Doesn't follow PEP8 indentation

File: app/api/__init__.py (hunk 1 above)
Comment (shubham-padia) [inline app/api/__init__.py]: No. I've used id in https://github.com/fossasia/open-event-orga-server/pull/3713/files#diff-b38e5fc3b36c615183eeb17dd03f0317R32 so this should be id.

File: app/api/events.py (hunk 2 above)
Comment (shubham-padia) [inline app/api/events.py]: done. Thanks

File: app/api/__init__.py (hunk 1 above)
Comment (poush) [inline app/api/__init__.py]: Please correct if I'm wrong here.
According to the docs if I see 
```api.route(PersonDetail, 'person_detail', '/persons/<int:id>', '/computers/<int:computer_id>/owner')```

What I am trying to do here is to get Owner( which is a person ) of a specific. Now to get this I will define a method. I will define a method ``` def before_get_object(self, view_kwargs):``` to get the related id of Person with that specific computer. 
Here are you saying it just as ```id``` which could be interpreted as event's id and there is no such addition of this in ``` def before_get_object``` function EventDetail.

I tried this with sponsors and at this endpoint (```/sponsors/1/event```) and I got the details of the event with ID 1 but the sponsor with id 1 was associated with event with id 2

File: app/api/tracks.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    schema = TrackSchema
+    data_layer = {'session': db.session,
+                  'model': Track,
+                  }
+
+
+class TrackDetail(ResourceDetail):
+    """
+    Track detail by id
+    """
```
Comment (poush) [inline app/api/tracks.py]: As mentioned above, there should be way to filter track's id from session's id which will passed when ```/sessions/<int:id>/track```

File: app/api/__init__.py (hunk 1 above)
Comment (shubham-padia) [inline app/api/__init__.py]: Yeah. You're right, updated it. Thanks :smile: .

File: app/api/tracks.py (hunk 3 above)
Comment (shubham-padia) [inline app/api/tracks.py]: Changes made

File: app/api/events.py [hunk 4]
```diff
@@ -45,11 +46,19 @@ class Meta:
...
+                          related_view_kwargs={'event_id': '<id>'},
+                          schema='TrackSchema',
+                          many=True,
+                          type_='track')
 
     sponsor = Relationship(attribute='sponsor',
                            self_view='v1.event_sponsor',
                            self_view_kwargs={'id': '<id>'},
-                           related_view='v1.sponsor_detail',
+                           related_view='v1.sponsor_list',
```
Comment (shubham-padia) [inline app/api/events.py]: I was getting an error because of this, this is a one to many relationship, thats why the change

File: app/api/__init__.py [hunk 5]
```diff
@@ -43,11 +45,12 @@
           '/microlocations/<int:id>/relationships/event')
 
 # sessions
-api.route(SessionList, 'session_list', '/sessions', '/events/<int:id>/sessions')
+api.route(SessionList, 'session_list', '/sessions', '/events/<int:id>/sessions', '/tracks/<int:track_id>/sessions')
```
Comment (poush) [inline app/api/__init__.py]: For this, you have to create one more method, similar to this   
```python
def query(self, view_kwargs):
        query_ = self.session.query(Sponsor)
        if view_kwargs.get('event_id') is not None:
            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
        return query_
```
This was for */events/<int:event_id>/sponsors*

Comment (shubham-padia) [inline app/api/__init__.py]: Thanks. Changes made.

File: app/api/events.py (hunk 4 above)
Comment (shubham-padia) [inline app/api/events.py]: Thanks. Changes made.
